test/integration/securetls_fips_test.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2021-2022 the Pinniped contributors. All Rights Reserved.
```

Comment:
I feel like there is a lot of duplication in this test that may be avoidable by splitting out the bits that change when under FIPS vs. not.  I am mostly wanting to avoid the case where we add tests to one file and fail to add them to the other.